### DISCUSSION TITLE
Java 11 or Java 14?

### DISCUSSION THREAD

**Discussion by CoffeeAddictAnonymous** [mars-sim/mars-sim#323] [Q&A] [answered]
I see on the main page its Java 11 from "latest" version onward. But I compressed the github files, and I see all references are to Java 14. Should I switch to that or stay with Java 11?

**Comment by bevans2000** [comment, marked as answer]
Still with Java 11.

**Reply by CoffeeAddictAnonymous** [reply]
Thanks for the confirmation :)

**Reply by mokun** [reply]
> Should I switch to that or stay with Java 11?

The short answer is that it doesn't matter.

I myself have been using Java 14 and 15 in Eclipse on my machine for a while and see no problem running mars-sim on top of either of them.

It's just that when it comes to compiling the mars-sim, we do it in Java 11 to ensure that those who have yet adopted Java 14/15 can still run it.

**Reply by mokun** [reply]
Would you be interested in testing/developing mars-sim by any chance in near future ?

We would usually have multiple versions of Java installed in our machines and from time to time, we switch to the latest stable one to just see ahead if there's any compatibility issues, you know.
